src/cmd/dev.go

```diff
@@ -79,11 +89,31 @@ var devDeployCmd = &cobra.Command{
 	},
 }
 
+// isLocalBundle checks if the bundle source is a local bundle
+func isLocalBundle(src string) bool {
+	return helpers.IsDir(src) || strings.Contains(src, ".tar.zst")
+}
+
+// validateDevDeployFlags validates the flags for dev deploy
+func validateDevDeployFlags(isLocalBundle bool) error {
+	if !isLocalBundle {
+		//Throw error if trying to run with --flavor, --flavor-all, or --force-create flag with remote bundle
+		if len(bundleCfg.DevDeployOpts.Flavor) > 0 || bundleCfg.DevDeployOpts.FlavorAll != "" || bundleCfg.DevDeployOpts.ForceCreate {
+			return fmt.Errorf("Cannot use --flavor, --flavor-all, or --force-create flags with remote bundle")
+		}
+	}
+	return nil
+}
+
 func init() {
 	initViper()
 	rootCmd.AddCommand(devCmd)
 	devCmd.AddCommand(devDeployCmd)
 	devDeployCmd.Flags().StringArrayVarP(&bundleCfg.DeployOpts.Packages, "packages", "p", []string{}, lang.CmdBundleDeployFlagPackages)
+	devDeployCmd.Flags().StringToStringVarP(&bundleCfg.DevDeployOpts.Ref, "ref", "r", map[string]string{}, lang.CmdBundleDeployFlagRef)
+	devDeployCmd.Flags().StringToStringVarP(&bundleCfg.DevDeployOpts.Flavor, "flavor", "f", map[string]string{}, lang.CmdBundleCreateFlagFlavor)
+	devDeployCmd.Flags().StringVar(&bundleCfg.DevDeployOpts.FlavorAll, "flavor-all", "", lang.CmdBundleCreateFlagFlavorAll)
```

Comment:
Instead of `--flavor-all`, what do you think of having just `--flavor` and inspecting the value to see if it should be applied to all packages or not? For example:
- `uds dev deploy ... --flavor ib` would apply to all packages in the bundle
- `uds dev deploy ... --flavor fooPkg=ib` would apply to only the "fooPkg` package

Reply:
yeah, I liked that better and it's what I had before, but cobra doesn't like it since it's expecting a map/entry. It wants a key=value

Reply:
Yeah we could get fancy and have Cobra expect a string and do the parsing ourselves, but I'm not sure if the juice is worth the squeeze, `--flavor-all` is fine for now